### PR TITLE
refactor: pleg/getContainersFromPods

### PR DESCRIPTION
**Pull request by Monokaix**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Do some clean and refactor code.
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: pkg/kubelet/pleg/generic.go [hunk 1]
```diff
@@ -325,29 +325,24 @@ func (g *GenericPLEG) relist() {
...
 			containers = append(containers, c)
 		}
+	}
 
+	for _, p := range pods {
+		if p == nil {
+			continue
+		}
+		fillCidSet(p.Containers)
+		// Update sandboxes as containers
```
Comment (endocrimes) [inline pkg/kubelet/pleg/generic.go:344]: maintain the TODO for sandbox tracking please :)

**Review by endocrimes — CHANGES_REQUESTED** [review]

File: pkg/kubelet/pleg/generic.go (hunk 1 above)
Comment (Monokaix) [reply pkg/kubelet/pleg/generic.go:344]: Thanks for reminding! done for that.

**Comment by haircommander** [comment]
/ok-to-test
/lgtm

a clean refactor! thanks @Monokaix

**Review by mrunalp — APPROVED** [review]

**Pull request merged**